File: day1.ts
const input: number[] = (await Deno.readTextFile("./data/day1.txt"))
  .split("\n")
  .map((s) => Number(s));

const targetSum = 2020;

for (let j = 0; j < input.length; j++) {
  const _targetSum = targetSum - input[j];
  const cache: number[] = [];

  for (let i = 0; i < input.length; i++) {
    const element = input[i];

    if (cache.includes(_targetSum - element)) {
      console.log(input[j] * element * (_targetSum - element));
      Deno.exit(1);
    } else {
      cache.push(element);
    }
  }
}

export {};

File: src/day1.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>

#define INPUT_LINES 200
#define TARGET_SUM 2020

void load_input();
bool includes(int, int);

int input[INPUT_LINES];
int cache[INPUT_LINES];

int main() {
  load_input();

  for (int i = 0; i < INPUT_LINES - 2; i++) {
    int target_sum_i = TARGET_SUM - input[i];
    int cache_index = -1;

    for (int j = i + 1; j < INPUT_LINES; j++) {
      int target_sum_j = target_sum_i - input[j];

      if (includes(cache_index + 1, target_sum_j)) {
        printf("%d\n", input[i] * input[j] * target_sum_j);
        return 0;
      } else {
        cache[++cache_index] = input[j];
      }
    }
  }

  return 0;
}

bool includes(int lim, int value) {
  for (int i = 0; i < lim; i++) {
    if (cache[i] == value) {
      return true;
    }
  }

  return false;
}

void load_input() {
  FILE *file = fopen("../input/day1.txt", "r");

  for (int i = 0; i < INPUT_LINES; i++) {
    fscanf(file, "%d", &input[i]);
  }
}
